libs/gl-client/src/signer/mod.rs

```diff
@@ -539,15 +539,24 @@ impl Signer {
                     .for_new_client(1 as u64, pk, c.dbid)
                     .handle(msg)
             }
+        };
+
+        if let Err(e) = response {
+            report::Reporter::report(crate::pb::scheduler::SignerRejection {
+                msg: format!("{:?}", e),
+                request: Some(req.clone()),
+                git_version: GITHASH.to_string(),
+            })
+            .await;
+            return Err(Error::Other(anyhow!("processing request: {e:?}")));
         }
-        .map_err(|e| Error::Other(anyhow!("processing request: {e:?}")))?;
 
+        let response = response.unwrap();
```

Comment:
You can avoid the clone by dropping the reporter call into the existing map_err closure:
```rust
.map_err(|e| 
    {
        report::Reporter::report(crate::pb::scheduler::SignerRejection {
                msg: format!("{:?}", e),
                request: Some(req.clone()),
                git_version: GITHASH.to_string(),
            })
            .await;
          Error::Other(anyhow!("processing request: {e:?}"))
    }
)?;

```

Reply:
Clever ^^

Reply:
Yeah, sadly we can't `.await` inside of a `map_err`, but would have been nice ^^

Reply:
Ahh, my async is rusty